File: Java_Virtual_World/src/Gameplay/Plants/Plant.java
package Gameplay.Plants;

import Gameplay.Organism;
import java.lang.Math;
import java.util.Vector;

public abstract class Plant extends Organism {
    static final int INITIATIVE = 0;
    static final double PROBABILITY = 0.1;

    public Plant(int x, int y, int s){
        super(x, y, s, INITIATIVE);
    }

    protected void spread(){
        if(Math.random() < PROBABILITY){
            Vector<Integer> tmp = world.getFreeSquare(x, y);
            if(!(tmp.get(0) == x && tmp.get(1) == y)){
                Organism o = Copy();
                o.setAge(0);
                o.setX(tmp.get(0));
                o.setY(tmp.get(1));
                world.addOrg(o);
            }
        }
    }

    @Override
    public void Action(){
        spread();
    }

    @Override
    public void Collision(){

    }
    @Override
    public void NextRound(){

    }
}

File: Python_Virtual_World/Plant.py
import copy
from random import random

from MyVector import MyVector
from Organism import Organism


class Plant(Organism):

    INITIATIVE = 0
    PROPABILITY = 0.05

    def __init__(self, position: MyVector, s: int):
        super().__init__(position, s, Plant.INITIATIVE)

    def action(self):

        self.spread()

    def collision(self):

        pass

    def newRound(self):

        pass


    def spread(self):

        if random() >= Plant.PROPABILITY:
            return

        newField = self._world.getFreeRandomSquare(self.getPosition())

        if newField is None:
            return

        org = copy.deepcopy(self)

        org.setAge(0)
        org.setPosition(newField)

        self._world.addOrg(org)
        self._world.getNots().add(self.__str__() + " has spreaded!")




